compiler/rustc_typeck/src/collect.rs

```diff
@@ -2784,6 +2784,10 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
                     // deterministic trap. There is no undefined behavior when
                     // executing WebAssembly so `#[target_feature]` is allowed
                     // on safe functions (but again, only for WebAssembly)
+                    //
+                    // Note that this is also allowed if `actually_rustdoc` so
+                    // if a target is documenting some wasm-specific code then
+                    // it's not spuriously denied.
```

Comment:
I really am not a fan of continuing to special-case rustdoc ... is there any way to avoid calling `codegen_fn_attrs` in rustdoc itself? Rustdoc shouldn't care what they are, right?

Reply:
No, the `codegen_fn_attrs` query is forced in the check attr pass which ensures that many attributes are valid. Otherwise `cargo check` would completely ignore it and `cargo build` would ignore it for functions that aren't codegened.